source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -152,6 +153,11 @@ Config::SharedConfig::SharedConfig(
     on_demand_config_ =
         std::make_unique<OnDemandConfig>(config.on_demand(), context, *stats_scope_);
   }
+
+  if (!config.proxy_protocol_tlvs().empty()) {
+    proxy_protocol_tlvs_ =
+        Extensions::Common::ProxyProtocol::parseTLVs(config.proxy_protocol_tlvs());
```

Comment:
Generally we try pretty hard to not add deps from core to extensions.  Would it be possible to instead configure which TLVs we put into ProxyProtocolFilterState, and then simply copy all when we hit the tcp proxy session?
/wait-any

Reply:
I'm not sure if I understand correctly, I believe this implementation does the same as you defined:
* Configures TLVs to be populated in the proxy protocol filter state in the config
* Copies the TLVs on every tcp-proxy session